src/Language/PureScript/Bundle.hs

```diff
@@ -567,13 +535,23 @@ codeGen optionsMainModule optionsNamespace ms =  renderToString (JSAstProgram (p
 -- | The bundling function.
 -- This function performs dead code elimination, filters empty modules
 -- and generates and prints the final Javascript bundle.
-bundle :: (MonadError ErrorMessage m)
-       => [(ModuleIdentifier, String)] -- ^ The input modules.  Each module should be javascript rendered from 'Language.PureScript.Make' or @psc@.
-       -> [ModuleIdentifier] -- ^ Entry points.  These module identifiers are used as the roots for dead-code elimination
-       -> Maybe String -- ^ An optional main module.
-       -> String -- ^ The namespace (e.g. PS).
-       -> m String
-bundle inputStrs entryPoints mainModule namespace = do
+bundle :: (Applicative m, MonadError ErrorMessage m)
+     => [(ModuleIdentifier, String)] -- ^ The input modules.  Each module should be javascript rendered from 'Language.PureScript.Make' or @psc@.
+     -> [ModuleIdentifier] -- ^ Entry points.  These module identifiers are used as the roots for dead-code elimination
+     -> Maybe String -- ^ An optional main
+     -> String -- ^ The namespace (e.g. PS).
+     -> Maybe String
+     -> m String
+bundle inputStrs entryPoints mainModule namespace optimize = do
+  let shouldUncurry = case optimize of
+                        Nothing -> False
+                        Just "uncurry" -> True
+                        Just "u" -> True
+                        Just "all" -> True
+                        Just "a" -> True
+                        Just _ -> False
+      secondRun = True
```

Comment:
I would still prefer to add an options record, since we'll probably end up adding more parameters this way, and it helps code comprehension.